discovery-provider/src/tasks/index_spl_token.py

```diff
@@ -126,6 +152,63 @@ def parse_spl_token_transaction(
         raise e
 
 
+def process_spl_token_transactions(
+    txs: List[ConfirmedTransaction], session
+) -> List[AudioTransactionsHistory]:
+    try:
+        audio_txs = []
+        for tx_info in txs:
+            logger.info(
+                f"index_spl_token.py | processing transaction: {tx_info['signature']} | slot={tx_info['slot']}"
+            )
+            # Index as an external receive transaction
+            # Note: external sends are under a different program, see index_user_bank.py
+            if tx_info["memo"] and "In-App $AUDIO Purchase:" not in tx_info["memo"]:
+                audio_txs.append(
+                    AudioTransactionsHistory(
+                        user_bank=tx_info["user_bank"],
+                        slot=tx_info["slot"],
+                        signature=tx_info["signature"],
+                        transaction_type=(TransactionType.transfer),
+                        method=TransactionMethod.receive,
+                        transaction_created_at=tx_info["timestamp"],
+                        change=Decimal(tx_info["postbalance"] - tx_info["prebalance"]),
+                        balance=Decimal(tx_info["postbalance"]),
+                        tx_metadata=tx_info["sender_wallet"],
+                    )
+                )
+
+            # Index as purchase transaction
+            else:
+                vendor = tx_info["memo"][1:-1].split(":")[1][1:]
+                if vendor not in purchase_vendor_map.keys():
+                    raise Exception(
+                        f"Purchase instruction from unsupported vendor: {vendor}"
+                    )
```

Comment:
do we definitely want to throw an exception here or just skip?

Reply:
I thought it might be `Unknown` in this case

Reply:
I think `Unknown` is for a specific rare case where in the buy audio recovery flow we can't recover the vendor cc @rickyrombo.
Down to skip, original intention was that if we get an unexpected value we probably want to know about it. But I think we should validate the memo earlier as your other comments suggest.